app/scripts/modules/core/src/cluster/AllClustersGroupings.tsx

```diff
@@ -13,71 +18,106 @@ export interface IAllClustersGroupingsProps {
 }
 
 export interface IAllClustersGroupingsState {
-  groups: IClusterGroup[];
+  groups: IClusterSubgroup[];
   sortFilter: any;
 }
 
+@UIRouterContext
+@BindAll()
 export class AllClustersGroupings extends React.Component<IAllClustersGroupingsProps, IAllClustersGroupingsState> {
   private clusterFilterService = ReactInjector.clusterFilterService;
   private clusterFilterModel = ReactInjector.clusterFilterModel;
 
   private groupsSubscription: Subscription;
   private unwatchSortFilter: Function;
 
+  private cellCache: CellMeasurerCache;
+
   constructor(props: IAllClustersGroupingsProps) {
     super(props);
+    this.cellCache = new CellMeasurerCache({ fixedWidth: true, minHeight: 100, defaultHeight: 177 });
     this.state = {
-      groups: this.clusterFilterModel.asFilterModel.groups,
+      groups: this.clusterFilterModel.asFilterModel.groups.reduce((a, b) => a.concat(b.subgroups), []),
       sortFilter: this.clusterFilterModel.asFilterModel.sortFilter,
     };
   }
 
   public componentDidMount() {
-    const onGroupsChanged = (groups: IClusterGroup[]) => this.setState({ groups });
+    const onGroupsChanged = (groups: IClusterGroup[]) => this.setState({ groups: groups.reduce((a, b) => a.concat(b.subgroups), []) });
     this.groupsSubscription = this.clusterFilterService.groupsUpdatedStream.subscribe(onGroupsChanged);
 
     const getSortFilter = () => this.clusterFilterModel.asFilterModel.sortFilter;
-    const onFilterChanged = ({ ...sortFilter }: any) => this.setState({ sortFilter });
+    const onFilterChanged = ({ ...sortFilter }: any) => {
+      this.setState({ sortFilter }, () => this.cellCache.clearAll());
+    };
     // TODO: Remove $rootScope. Keeping it here so we can use $watch for now.
     //       Eventually, there should be events fired when filters change.
     this.unwatchSortFilter = ReactInjector.$rootScope.$watch(getSortFilter, onFilterChanged, true);
   }
 
+  public componentWillReceiveProps() {
+    this.cellCache.clearAll();
+  }
+
   public componentWillUnmount() {
     this.groupsSubscription.unsubscribe();
     this.unwatchSortFilter();
   }
 
-  public render() {
-    const { app, initialized } = this.props;
+  private renderRow(props: ListRowProps): JSX.Element {
+    const { app } = this.props;
     const { groups = [], sortFilter } = this.state;
+    const group = groups[props.index];
+    const parent: React.ReactType = props.parent as any; // bad cast, sad @types
+    return (
+      <CellMeasurer
+        cache={this.cellCache}
+        key={props.key}
+        rowIndex={props.index}
+        parent={parent}
+      >
+        <div key={props.index} style={props.style}>
+          <div>
```

Comment:
this <div> is not necessary

Reply:
Whoops, you are right. This was an artifact of debugging issues with the CellMeasurer. Will remove in a subsequent PR